zetaclient/evm_client.go

```diff
@@ -799,12 +808,23 @@ func (ob *EVMChainClient) observeInTX() error {
 		if !ob.chain.IsKlaytnChain() {
 			for bn := startBlock; bn <= toBlock; bn++ {
 				//block, err := ob.EvmClient.BlockByNumber(context.Background(), big.NewInt(int64(bn)))
-				block, err := ob.EvmClient.BlockByNumber(context.Background(), big.NewInt(bn))
+				block, err := ob.GetBlockByNumberCached(bn)
 				if err != nil {
 					ob.logger.ExternalChainWatcher.Error().Err(err).Msgf("error getting block: %d", bn)
 					continue
 				}
-				//ob.logger.ExternalChainWatcher.Debug().Msgf("block %d: num txs: %d", bn, len(block.Transactions()))
+				_ = ob.BlockCache.Add(block.Hash(), block)
+				headerRLP, err := rlp.EncodeToBytes(block.Header())
+				if err != nil {
+					ob.logger.ExternalChainWatcher.Error().Err(err).Msgf("error encoding block header: %d", bn)
+					continue
+				}
+				_, err = ob.zetaClient.PostAddBlockHeader(ob.chain.ChainId, block.Hash().Bytes(), block.Number().Int64(), headerRLP)
+				if err != nil {
+					ob.logger.ExternalChainWatcher.Error().Err(err).Msgf("error posting block header: %d", bn)
+					continue
```

Comment:
I think this newly added `continue` will potentially disrupt, on error, execution of following logics. To play safe, we can just log it and remove the `continue`

Reply:
let's leave it to another issue/PR
this is part of a bigger problem:
https://github.com/zeta-chain/node/issues/1133